pkg/gitrepo/repo.go

```diff
@@ -303,6 +305,39 @@ func (r *Repo) ResolveVersion(ctx context.Context, ref string) (string, error) {
 	return pseudoVersion, nil
 }
 
+// GetFile retrieves content of file stored at path on the master branch.
+func (r *Repo) GetFileContent(path string) ([]byte, error) {
+	repo, err := git.Open(r.storage, r.worktree)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// When empty CheckoutOptions defaults to master branch.
+	err = worktree.Checkout(&git.CheckoutOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
Do we need to checkout here?

Reply:
Yes otherwise the worktree is empty and we cannot open the file. Also I was thinking it would be good to expose the checkout options, so we can retrieve a file at a specific version and not just master. This would also improves the testing as the file at a specific version should never change.

Reply:
I agree exposing CheckoutOptions makes sense. It also creates the notion of performing this action by the method. But let's please create our own struct here to not leak external import. I think branch name would be sufficient for now.

Reply:
done. PTALA